src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -403,15 +384,16 @@ public int GroupNumberFromName(string name)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.name);
             }
 
-            if (_srm is not null)
+            if ((roptions & RegexOptions.NonBacktracking) != 0)
             {
                 // In NonBacktracking mode there is a single top level group 0
+                // See comment in GetGroupNames.
```

Comment:
should this magic 0 value be a const field maybe?

Reply:
The top level group is always 0 and "0" as a string. Not sure these need separate constants.